mk/config.mk

```diff
@@ -725,3 +725,17 @@ CFG_DRIVERS_CLK_FIXED ?= $(CFG_DRIVERS_CLK_DT)
 
 $(eval $(call cfg-depends-all,CFG_DRIVERS_CLK_DT,CFG_DRIVERS_CLK CFG_DT))
 $(eval $(call cfg-depends-all,CFG_DRIVERS_CLK_FIXED,CFG_DRIVERS_CLK_DT))
+
+# Branch Target Identification (part of the ARMv8.5 Extensions) provides a
+# mechanism to limit the set of locations to which computed branch instructions
+# such as BR or BLR can jump. To make use of BTI in TEE cores and ldelf on CPU's
+# that support it, enable this option.
+CFG_CORE_BTI	?= n
+
+$(eval $(call cfg-depends-all,CFG_CORE_BTI,CFG_ARM64_core))
+
+# To make use of BTI in user space libraries and TA's on CPU's that support it,
+# enable this option
+CFG_TA_BTI	?= n
+
+$(eval $(call cfg-depends-all,CFG_TA_BTI,CFG_ARM64_core))
```

Comment:
This will change `CFG_TA_BTI` from `y` to `n` if not a 64b core target, with a nice build trace.
I wonder whether a build error is preferred: config expected BTI but binary won't support it and one may not notice that unless one tracks build traces.
This may not be an issue in this P-R but rather a generic issue in `cfg-depends-all`. Should this macro abort the build sequence with an explicit error, rather than disabling the configuration switch?

Reply:
> This may not be an issue in this P-R but rather a generic issue in `cfg-depends-all`. Should this macro abort the build sequence with an explicit error, rather than disabling the configuration switch?

I tend to agree, but it needs a bit more thoughts because we do have a few of these warnings in the CI build ("core/crypto.mk:108: Warning: Enabling CFG_CRYPTO_AES [required by CFG_REE_FS]) and when I tried fixing them by adding `CFG_REE_FS=n` I got a build error in LibTomCrypt :-/ I'll take a closer look.